src/execution_plan/execution_plan_reduce_to_apply.c

```diff
@@ -0,0 +1,130 @@
+ * filter tree, and checks if if the current subtree rooted at the visited node contains path filter or not,
+ * and either reduces the root or continue traversal and reduction.
+ * The three possible operations could be:
+ * 1. OpFilter - In the case the subtree originated from the root does not contains any path filters.
+ * 2. OpSemiApply - In case of the current root is an expression which contains a path filter.
+ * 3. ApplyMultiplexer - In case the current root is an operator (OR or AND) and at least one of its children
+ * has been reduced to OpSemiApply or ApplyMultiplexer. */
+static OpBase *_ReduceFilterToOpBase(ExecutionPlan *plan,
+									 FT_FilterNode *filter_root) {
+	FT_FilterNode *node;
+	// Case of an expression which contains path filter.
+	if(filter_root->t == FT_N_EXP && FilterTree_ContainsFunc(filter_root, "path_filter", &node)) {
+		AR_ExpNode *expression = filter_root->exp.exp;
+		bool anti = false;
+		if(strcasecmp(expression->op.func_name, "not") == 0) {
+			anti = true;
+			expression = expression->op.children[0];
+		}
+		// Build new Semi Apply op.
+		OpBase *op_semi_apply = NewSemiApplyOp(plan, anti);
+		const cypher_astnode_t *path = expression->op.children[0]->operand.constant.ptrval;
+		// Add a match branch as a Semi Apply op child.
+		ExecutionPlan_AddOp(op_semi_apply, _buildMatchBranch(plan, path));
+		return op_semi_apply;
+	}
+	// Case of an operator (Or or And) which its subtree contains path filter
+	if(filter_root->t == FT_N_COND && FilterTree_ContainsFunc(filter_root, "path_filter", &node)) {
+		// Create the relevant LHS branch and set a bounded branch for it.
+		OpBase *lhs = _ReduceFilterToOpBase(plan, filter_root->cond.left);
+		if(lhs->type == OPType_FILTER) filter_root->cond.left = NULL;
+		_CreateBoundBranch(lhs, plan);
+		// Create the relevant RHS branch and set a bounded branch for it.
+		OpBase *rhs = _ReduceFilterToOpBase(plan, filter_root->cond.right);
+		if(rhs->type == OPType_FILTER) filter_root->cond.right = NULL;
+		_CreateBoundBranch(rhs, plan);
+		// Create multiplexer op and set the branches as its children.
+		OpBase *apply_multiplexer = NewApplyMultiplexerOp(plan, filter_root->cond.op);
+		ExecutionPlan_AddOp(apply_multiplexer, lhs);
+		ExecutionPlan_AddOp(apply_multiplexer, rhs);
+		return apply_multiplexer;
+	}
+	// In the case of a simple filter.
+	return NewFilterOp(plan, filter_root);
+}
+
+/* Reduces a filter operation to a SemiApply/ApplyMultiplexer operation, according to the filter tree */
+void ExecutionPlan_ReduceFilterToApply(ExecutionPlan *plan, OpBase *op) {
+	assert(op->type == OPType_FILTER);
+	OpFilter *filter = (OpFilter *) op;
+	// Reduce.
+	OpBase *apply_op = _ReduceFilterToOpBase(plan, filter->filterTree);
+	// Replace operations.
+	ExecutionPlan_ReplaceOp(plan, op, apply_op);
+	// Bounded branch is now the last child (after ops replacement). Make it the first.
+	_OpBaseSwapChildren(apply_op, 0, apply_op->childCount - 1);
```

Comment:
Did we introduce `_OpBaseSwapChildren` just for this part?
if so please remove it and introduce logic within the apply operation to sort its streams.